### PR TITLE
Added support to pass dimensions on filter submit endpoint

### PR DESCRIPTION
**Pull request by j-s-rawat**
### What
US: https://trello.com/c/vSxzIedn/5674-update-dp-cantabular-metadata-exporter-to-support-export-based-on-specific-dimensions
US: https://trello.com/c/fXFPiqZF/5691-update-populationtype-with-valid-data-when-filter-is-submitted

- Added support for dimensions to be passed in kafka ExportStartEvent message
- Added population type for filter output

### How to review

New structure of event matches with the event used in CSV exporter. There is a PR for that as well
PR: https://github.com/ONSdigital/dp-cantabular-csv-exporter/pull/67

### Who can review
Anyone

### CODE REVIEW

File: features/submit_filter.feature [hunk 1]
```diff
@@ -180,9 +180,10 @@ Feature: Submit Filter Private Endpoints Not Enabled
...
-      "type": "flexible"
+      "type": "flexible",
+      "population_type": "Example"
     }
     """
     And the following Export Start events are produced:
-      | InstanceID        | DatasetID            | Edition          | Version | FilterOutputID                       |
-      | TEST-INSTANCE-ID  | cantabular-example-1 | 2021             | 1       | 94310d8d-72d6-492a-bc30-27584627edb1 |
+      | InstanceID        | DatasetID            | Edition          | Version | FilterOutputID                       | DimensionIDs |
+      | TEST-INSTANCE-ID  | cantabular-example-1 | 2021             | 1       | 94310d8d-72d6-492a-bc30-27584627edb1 | Number of siblings (3 mappings),City|
```
Comment (ChrysmOre) [inline features/submit_filter.feature:189]: `Number Of siblings (3 mappings)` and `City` are lables not ids. The ids would be `siblings_3` and `city` probably best to be accurate here

**Review by ChrysmOre — COMMENTED** [review]
Small comment, also branch is out of date

File: features/submit_filter.feature (hunk 1 above)
Comment (j-s-rawat) [reply features/submit_filter.feature:189]: resolved

File: api/filters.go [hunk 2]
```diff
@@ -178,16 +179,20 @@ func (api *API) submitFilter(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	var dIDs []string
+	for _, d := range filter.Dimensions {
+		dIDs = append(dIDs, d.Name)
```
Comment (ChrysmOre) [inline api/filters.go:184]: Are we using dimension names or dimension IDs? We need to be clear and keep the naming accurate

File: features/steps/steps.go [hunk 3]
```diff
@@ -140,8 +141,23 @@ func (c *Component) iShouldReceiveAnErrorsArray() error {
...
+		return []string{}, nil
+	}
+
+	strArray := strings.Split(str, ",")
+	return strArray, nil
+}
+
 func (c *Component) theFollowingExportStartEventsAreProduced(events *godog.Table) error {
-	expected, err := assistdog.NewDefault().CreateSlice(new(event.ExportStart), events)
+	assist := assistdog.NewDefault()
```
Comment (saminahbab) [inline features/steps/steps.go:158]: Is there a slightly better place to put this kind of configuration? 
This is more related to test setup, and not step definition

**Review by saminahbab — DISMISSED** [review]

**Pull request merged**